kt/godot-library/src/main/kotlin/godot/core/memory/MemoryManager.kt

```diff
@@ -67,6 +68,24 @@ internal object MemoryManager {
 
     private var gcState = GCState.NONE
 
+    // A basic LRU cache.
+    private class LRUCache<K, V>(private val capacity: Int) : LinkedHashMap<K, V>(capacity, 0.75f, true) {
+        override fun removeEldestEntry(eldest: MutableMap.MutableEntry<K, V>?): Boolean {
+            return size > capacity
+        }
+    }
+
+    // Create an LRU cache for StringName objects based on a String key.
+    private val stringNameCache = LRUCache<String, StringName>(100)
```

Comment:
the size needs to be definable through commandline args. but as this requires cpp changes and jni calls it's fine for me if we do this in another pr if you don't want to do this.
Just tell me and I'll create a separate issue for this.

Reply:
I've added a TODO comment for now, let me take a look at the example you referenced and see if I can copy the implementation.